### PR TITLE
xtl: New package

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: X/xtl/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+cmake --install build
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [AnyPlatform()]
+
+# The products that we will ensure are always built
+products = Product[
+    # This is a header-only library without any products
```
Comment (giordano) [inline X/xtl/build_tarballs.jl:30]: Could list some header files as `FileProduct`s?  That has the benefit of checking that the files actually exist.

File: X/xtl/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
```
Comment (giordano) [inline X/xtl/build_tarballs.jl:34]:
```suggestion
dependencies = Dependency[
```

File: X/xtl/build_tarballs.jl (hunk 1 above)
Comment (eschnett) [reply X/xtl/build_tarballs.jl]: Done.

File: X/xtl/build_tarballs.jl (hunk 2 above)
Comment (eschnett) [reply X/xtl/build_tarballs.jl]: Done.

**Pull request merged**